session-10/src/components/Header/Header.js

```diff
@@ -0,0 +1,38 @@
+import React, { useRef, useState } from "react";
+import "./Header.css";
+import logo from "../../assets/logo.png";
+import profile from "../../assets/profile.png";
+import { useUserContext } from "../../context/UserContext/UserContext";
+import { useThemeContext } from "../../context/ThemeContext/ThemeContext";
+
+function Header() {
+  const { theme, setTheme } = useThemeContext();
+  const { reference, handleSearchText } = useUserContext();
+
+  const themeButton = theme == "light" ? "Dark" : "Light";
+
+  const toggleTheme = () => {
+    setTheme(theme == "light" ? "dark" : "light");
```

Comment:
make variables of `theme === "light"` and `theme === "dark"` on top. Use them throughout your code.